src/EditorFeatures/Text/Extensions.cs

```diff
@@ -37,7 +40,7 @@ public static SourceText AsText(this ITextSnapshot textSnapshot)
             => SnapshotSourceText.From(textSnapshot);
 
         internal static SourceText AsRoslynText(this ITextSnapshot textSnapshot, Encoding encoding)
-            => new SnapshotSourceText.ClosedSnapshotSourceText(textSnapshot, encoding);
+            => new SnapshotSourceText.ClosedSnapshotSourceText(((ITextSnapshot2)textSnapshot).TextImage, encoding);
```

Comment:
My worry was http://index/?leftProject=Microsoft.VisualStudio.LanguageServices&leftSymbol=x6edbstxxst3&file=Implementation%5CProjectSystem%5CVisualStudioWorkspaceImpl.cs&line=1256, where we actually process file opens in VS. Looks like we're good.